sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffledSizedHashJoinExec.scala

```diff
@@ -1023,15 +1023,26 @@ object GpuShuffledAsymmetricHashJoinExec {
       }
     }
 
-    private def getSingleBuildBatch(
+    private def getAsSingleBuildBatch(
         baseIter: Iterator[ColumnarBatch],
         exprs: BoundJoinExprs,
         metrics: Map[String, GpuMetric]): ColumnarBatch = {
       val iter = addNullFilterIfNecessary(baseIter, exprs.boundBuildKeys,
         exprs.buildSideNeedsNullFilter, metrics)
-      closeOnExcept(iter.next()) { batch =>
-        assert(!iter.hasNext)
-        batch
+      // Multiple small batches may exist when split-retry happens in the previous op.
+      // So need to concat them into a single one
+      val spBatches = mutable.Queue.empty[SpillableColumnarBatch]
+      closeOnExcept(spBatches) { _ =>
+        while(iter.hasNext) {
+          spBatches.enqueue(
+            SpillableColumnarBatch(iter.next(), SpillPriorities.ACTIVE_BATCHING_PRIORITY))
+        }
+      }
+      assert(spBatches.nonEmpty, "At least one batch is expected")
+      val cbTypes = spBatches.head.dataTypes
+      withRetryNoSplit(spBatches.toSeq) { _ =>
+        ConcatAndConsumeAll.buildNonEmptyBatchFromTypes(
+          spBatches.toArray.safeMap(_.getColumnarBatch()), cbTypes)
```

Comment:
no need to change anything in this PR but it
looks like this unnecessary conversion toArray is forced upon this callsite only because buildNonEmptyBatchFromTypes takes an Array. It could be changed to Seq which Array also is.